### ISSUE TITLE
Crash when change UIcontrols hierarchy on UI prefab

### ISSUE THREAD

**Issue by RuanLucasGD** [FlaxEngine/FlaxEngine#2082]
**Issue description:**
Prefab editor crash if i try change controls order and save.

**Steps to reproduce:**
1. Create a UI prefab
2. Create two Controls
3. Set the secondary UI control as child of the first UI control
4. Set the control types (In my case a Label and  a Border)
5. Save
6. Change the hierarchy childs horder
7. Save
8. Crash

**Minimal reproduction project:**

https://github.com/FlaxEngine/FlaxEngine/assets/79365912/a691ae21-713a-4660-8814-872ef332e650

**Flax version:**
1.7 master

**Comment by RuanLucasGD** [comment]
Error:

```
Stack overflow.
Repeat 130750 times:
--------------------------------
   at FlaxEngine.GUI.ContainerControl.LockChildrenRecursive()
--------------------------------
   at FlaxEngine.GUI.ContainerControl.PerformLayout(Boolean)
   at FlaxEngine.GUI.ContainerControl.OnChildrenChanged()
   at FlaxEngine.GUI.ContainerControl.AddChildInternal(FlaxEngine.GUI.Control)
   at FlaxEngine.GUI.Control.set_Parent(FlaxEngine.GUI.ContainerControl)
   at FlaxEngine.UIControl.set_Control(FlaxEngine.GUI.Control)
   at FlaxEngine.UIControl.Deserialize(System.String, System.Type)
   at FlaxEngine.Interop.NativeInterop+Invoker+InvokerNoRet2`3[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MarshalAndInvoke(System.Object, FlaxEngine.Interop.ManagedHandle, IntPtr)
   at FlaxEngine.Interop.NativeInterop.InvokeMethod(FlaxEngine.Interop.ManagedHandle, FlaxEngine.Interop.ManagedHandle, IntPtr, IntPtr)
2
   at FlaxEngine.PrefabManager.<Internal_ApplyAll>g____PInvoke|17_0(IntPtr)
   at FlaxEngine.PrefabManager.Internal_ApplyAll(IntPtr)
   at FlaxEditor.Modules.PrefabsModule.ApplyAll(FlaxEngine.Actor)
   at FlaxEditor.Windows.Assets.PrefabWindow.Save()
   at FlaxEditor.GUI.ToolStripButton.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
2
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEditor.Windows.Assets.PrefabWindow.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEditor.GUI.Docking.DockPanelProxy.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
2
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.GUI.WindowRootControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.Window.Internal_OnMouseUp(FlaxEngine.Float2 ByRef, FlaxEngine.MouseButton)
   at FlaxEngine.Interop.NativeInterop+Invoker+InvokerNoRet2`3[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[FlaxEngine.Float2, FlaxEngine.CSharp, Version=1.7.6406.1, Culture=neutral, PublicKeyToken=null],[FlaxEngine.MouseButton, FlaxEngine.CSharp, Version=1.7.6406.1, Culture=neutral, PublicKeyToken=null]].MarshalAndInvoke(System.Object, FlaxEngine.Interop.ManagedHandle, IntPtr)
   at FlaxEngine.Interop.NativeInterop.InvokeMethod(FlaxEngine.Interop.ManagedHandle, FlaxEngine.Interop.ManagedHandle, IntPtr, IntPtr)
```

**Comment by mafiesto4** [comment]
Fixed in e448692eb95b17d78c5e1a4c2472d72a75b40495.
